app/controllers/works_controller.rb

```diff
@@ -107,10 +109,16 @@ def set_content
     @work_form.content_id = @content.id
   end
 
+  def source_id
```

Comment:
Can we leave the source_id logic in the DepositJob? Seems like knowing about that should be its responsibility instead of the controller.